ckanext/ontario_theme/templates/internal/snippets/search_form.html

```diff
@@ -47,166 +50,213 @@ <h2 class="ontario-label">
+                       sol:category="primary"
+                       viewBox="0 0 24 24"
+                       preserveAspectRatio="xMidYMid meet">
+                    <use xlink:href="#ontario-icon-sort"></use>
+                  </svg>
+                {{ _('Sort') }}</span>
+                <span aria-label="{{ _('close') }}"
+                      class="sort-by-close"
+                      role="button"
+                      tabindex="0">
+                  <svg class="ontario-icon hide-sort-by"
+                       focusable="false"
+                       sol:category="primary"
+                       viewBox="0 0 24 24"
+                       preserveAspectRatio="xMidYMid meet">
+                    <use xlink:href="#ontario-icon-close"></use>
+                  </svg>
+                </span>
+              </div>
+              <ul>
+                {% for label, value in sorting %}
+                  {% set sort_in_params = true if request.params['sort'] %}
+                  {% if label and value %}
+                    {% set selected = 'selected' if (sorting_selected == value) %}
+                    {% set href = h.remove_url_param('sort', replace=value, extras=extras) if sort_in_params else h.add_url_param(new_params={'sort': value}, extras=extras) %}
+                    <li class="sort-by-options {{ selected }}">
+                      <a href="{{ href }}">{{ label }}</a>
+                    </li>
+                  {% endif %}
+                {% endfor %}
+              </ul>
             </div>
-            <ul>
+          </div>
+        {% endif %}
+        <div class="checkbox-status" role="status">
+          {% if facets %}
+            {% set border_highlight = "ontario-border-highlight--dark-blue" if count !=0 else "ontario-border-highlight--dark-taupe" %}
+            {% if count and query %}
+              {% set callout_text = _('Showing datasets found with search term <strong>"{query}"</strong> and access level') %}
+            {% elif not count and query %}
+              {% set callout_text = _('No datasets found with search term <strong>"{query}"</strong> and access level') %}
+            {% elif count %}
+              {% set callout_text = _("Showing datasets with access level") %}
+            {% else %}
+              {% set callout_text = _("No datasets with access level") %}
+            {% endif %}
+            <div class="ontario-callout {{ border_highlight }} access-level-status ontario-small-12 ontario-large-7 ontario-columns">
+              <p id="access-level-sentence-id">
+                {{ callout_text.format(query=query|escape)|safe }}{{ _(':') }} <span id="access-level-sentence-value"></span>
+              </p>
+            </div>
+          {% endif %}
+          <div class="form-select form-group control-order-by search-page-select ontario-large-5 ontario-columns">
+            <label class="ontario-label" for="field-order-by">{{ _('Sort by') + _(':') }}</label>
```

Comment:
When `count` = 0, the "Sort by' dropdown menu is still displayed on desktop screens, but on mobile the sort button is not displayed.